In [1]:
import pandas as pd
import numpy as np

In [2]:
# DE1_0_2008_Beneficiary_Summary_File_Sample_1.csv
# DE1_0_2008_to_2010_Carrier_Claims_Sample_1A.csv
# DE1_0_2008_to_2010_Carrier_Claims_Sample_1B.csv
# DE1_0_2008_to_2010_Inpatient_Claims_Sample_1.csv
# DE1_0_2008_to_2010_Outpatient_Claims_Sample_1.csv
# DE1_0_2008_to_2010_Prescription_Drug_Events_Sample_1.csv
# DE1_0_2009_Beneficiary_Summary_File_Sample_1.csv
# DE1_0_2010_Beneficiary_Summary_File_Sample_20.csv

In [5]:
datadir = '/home/xavier/data/cms_data/Data-Entrepreneurs-Synthetic-PUF/'
summaryFile = 'DE1_0_2008_Beneficiary_Summary_File_Sample_1.csv'
claimFile = 'DE1_0_2008_to_2010_Carrier_Claims_Sample_1A.csv'
drugFile = 'DE1_0_2008_to_2010_Prescription_Drug_Events_Sample_1.csv'

In [6]:
claims = pd.read_csv(datadir + drugFile)

/home/xavier/miniconda3/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (3) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [7]:
list(claims)

['DESYNPUF_ID',
 'PDE_ID',
 'SRVC_DT',
 'PROD_SRVC_ID',
 'QTY_DSPNSD_NUM',
 'DAYS_SUPLY_NUM',
 'PTNT_PAY_AMT',
 'TOT_RX_CST_AMT']

In [9]:
claims.head(15)

DESYNPUF_ID           PDE_ID   SRVC_DT PROD_SRVC_ID  QTY_DSPNSD_NUM  \
0   00013D2EFD8E45D1  233664490397622  20080103    247037252            30.0   
1   00013D2EFD8E45D1  233644490171972  20080105    223039502            10.0   
2   00013D2EFD8E45D1  233974489116848  20080109    364724812           120.0   
3   00013D2EFD8E45D1  233574491083209  20080123    179180672            30.0   
4   00013D2EFD8E45D1  233024491180571  20080124  58016005300            30.0   
5   00013D2EFD8E45D1  233884492803278  20080124  53650001801            30.0   
6   00013D2EFD8E45D1  233594491623171  20080125      8032506           180.0   
7   00013D2EFD8E45D1  233894491962732  20080129  58016081825           100.0   
8   00013D2EFD8E45D1  233344491666810  20080201  51129143701            30.0   
9   00013D2EFD8E45D1  233884490733605  20080205  67668013231            10.0   
10  00013D2EFD8E45D1  233254492355736  20080214    254511335            60.0   
11  00013D2EFD8E45D1  233784494196079  20080216  50070000265           360.0   
12  00013D2EFD8E45D1  233604491119946  20080223  53014072671            30.0   
13  00013D2EFD8E45D1  233404493740714  20080303  58016094835            30.0   
14  00013D2EFD8E45D1  233354489873968  20080305  64899011710            10.0   

    DAYS_SUPLY_NUM  PTNT_PAY_AMT  TOT_RX_CST_AMT  
0               20          10.0           120.0  
1               10           0.0             0.0  
2               30          10.0           110.0  
3               30           0.0           240.0  
4               30          70.0            70.0  
5               30          10.0            40.0  
6               30           0.0            10.0  
7               30          20.0            10.0  
8               30          10.0             0.0  
9               10          10.0            30.0  
10              30           0.0            10.0  
11              30           0.0            10.0  
12              30           0.0           150.0  
13              30           0.0            40.0  
14              30          10.0           230.0

In [7]:
icd = claims.loc[:,'DESYNPUF_ID':'ICD9_DGNS_CD_8']
icd.shape

(2370667, 12)

In [8]:
utibase = icd[
    icd['ICD9_DGNS_CD_1']
    .str.contains('599', case = False, na=False)]

In [9]:
utibase.shape

(7125, 12)

In [10]:
utibase

DESYNPUF_ID           CLM_ID  CLM_FROM_DT  CLM_THRU_DT  \
8        00016F745862898F  887953385848149     20081020     20081020   
413      0004F0ABD505251D  887593389474703     20091122     20091122   
577      00052705243EA128  887323388671747     20100409     20100409   
654      0007F12A492FD25D  887353385949566     20080620     20080620   
1227     0011CB1FE23E91AF  887343385957126     20080730     20080803   
1436     00139C345A104F72  887833387309417     20091105     20091105   
1850     0018C6D3871BA1E8  887663388998553     20080530     20080530   
2376     001EA2F4DB30F105  887843386072543     20091028     20091028   
2406     001EA2F4DB30F105  887163387079761     20100920     20100920   
2452     001F135A720283D1  887333388854561     20090130     20090130   
2483     001F135A720283D1  887863389371691     20091111     20091111   
2764     0021D4CDAFC0609F  887303386370784     20081028     20081028   
2977     002354398A00234E  887223388791946     20091127     20091127   
3323     0024E5A7B7272E75  887103385080144     20100421     20100421   
3348     00256A69373687B4  887033387421747     20080428     20080428   
3441     002578091E1953AF  887563388640427     20080313     20080313   
3515     002578091E1953AF  887873387995487     20091223     20091223   
3604     0025ACAB19F5DEC0  887193388527858     20090324     20090401   
4248     002AB71D3224BE66  887093386152471     20090815     20090823   
4286     002AB71D3224BE66  887393388083572     20100726     20100726   
4317     002AFD7289D200A7  887253389408040     20080913     20080913   
4322     002AFD7289D200A7  887473388089621     20081010     20081010   
4343     002AFD7289D200A7  887733386881483     20090725     20090725   
5654     0036004F5BAF9171  887193385256428     20100922     20100922   
5818     0038C1B1E637A1C7  887043386727457     20100717     20100717   
6486     0046020E72E1CBFD  887913386115032     20080831     20080831   
6508     0046020E72E1CBFD  887183388820748     20090324     20090324   
6783     004CA2ABE3A188A3  887653389178496     20091009     20091009   
7103     00534E60E969C69E  887613388302743     20080403     20080403   
8050     00604E938DB1DD50  887893387701012     20100801     20100801   
...                   ...              ...          ...          ...   
2360716  7E8C61B32BAFCC18  887893388439050     20100519     20100519   
2361038  7E9176265282601F  887483387524209     20081103     20081103   
2361217  7E932B592A36B3F2  887203385042278     20100709     20100709   
2361478  7E978C3144594CC2  887243385966210     20091218     20091218   
2361569  7E9875157A915F8A  887083386304533     20100213     20100213   
2362612  7EA6C2A3AC7650B8  887173386584580     20100312     20100312   
2363074  7EAEC15E06D9B685  887993385782038     20101012     20101012   
2363373  7EB4BD1BB601E201  887523385967880     20090102     20090102   
2363468  7EB6D72572D996DB  887463386650590     20091206     20091206   
2363855  7EBD1CA770D1215F  887453388090084     20090701     20090701   
2364135  7EC03B321A9E319F  887213388754063     20101103     20101103   
2364887  7EC637BC6A7EFF9E  887143384989589     20090115     20090117   
2365312  7ECE370A4F03BB55  887873385114592     20100211     20100211   
2365539  7ED2D662722FE5C3  887213388143889     20100216     20100216   
2365765  7ED5D270889285D9  887723385345772     20090618     20090618   
2365997  7ED9E4B16B37DDAC  887783388937021     20090912     20090912   
2366104  7EDD7AA045A115E5  887523384951443     20080905     20080905   
2367357  7EEE09EC27605873  887303388266352     20080127     20080127   
2367597  7EF3D29C645FF758  887093386485165     20090111     20090111   
2367737  7EF534D9A8000141  887743387774513     20090712     20090712   
2367779  7EF639D1344E5ED6  887963388352341     20080129     20080129   
2368096  7EFD0BC2354BB90D  887113388596111     20090331     20090331   
2368132  7EFD6371947E6B20  887193387952900     20080202     20080202   
2368427  7EFEE535C17CF51A  887903387227929     2009010

In [11]:
utifive = icd[
    icd['ICD9_DGNS_CD_1']
    .str.contains('^599', case = False, na=False)]
utifive.shape

(5574, 12)

In [12]:
utifive

DESYNPUF_ID           CLM_ID  CLM_FROM_DT  CLM_THRU_DT  \
8        00016F745862898F  887953385848149     20081020     20081020   
413      0004F0ABD505251D  887593389474703     20091122     20091122   
577      00052705243EA128  887323388671747     20100409     20100409   
654      0007F12A492FD25D  887353385949566     20080620     20080620   
1436     00139C345A104F72  887833387309417     20091105     20091105   
1850     0018C6D3871BA1E8  887663388998553     20080530     20080530   
2376     001EA2F4DB30F105  887843386072543     20091028     20091028   
2406     001EA2F4DB30F105  887163387079761     20100920     20100920   
2452     001F135A720283D1  887333388854561     20090130     20090130   
2483     001F135A720283D1  887863389371691     20091111     20091111   
2977     002354398A00234E  887223388791946     20091127     20091127   
3348     00256A69373687B4  887033387421747     20080428     20080428   
3441     002578091E1953AF  887563388640427     20080313     20080313   
3515     002578091E1953AF  887873387995487     20091223     20091223   
4248     002AB71D3224BE66  887093386152471     20090815     20090823   
4286     002AB71D3224BE66  887393388083572     20100726     20100726   
4322     002AFD7289D200A7  887473388089621     20081010     20081010   
4343     002AFD7289D200A7  887733386881483     20090725     20090725   
5654     0036004F5BAF9171  887193385256428     20100922     20100922   
5818     0038C1B1E637A1C7  887043386727457     20100717     20100717   
6486     0046020E72E1CBFD  887913386115032     20080831     20080831   
6508     0046020E72E1CBFD  887183388820748     20090324     20090324   
7103     00534E60E969C69E  887613388302743     20080403     20080403   
8050     00604E938DB1DD50  887893387701012     20100801     20100801   
8151     006084B3FA2A151C  887163385004200     20101227     20101227   
8303     0060C263DA4EBBAF  887913386460961     20080725     20080725   
9099     006AE3882B411074  887343385972791     20100127     20100127   
9401     00700D980231CC80  887163387474085     20091024     20091024   
9496     0070946DE04C2014  887743388296394     20080116     20080116   
9997     00765B329D6F7FDE  887713386183196     20081127     20081127   
...                   ...              ...          ...          ...   
2359064  7E74782C254F6B35  887473385301469     20080626     20080626   
2359073  7E74782C254F6B35  887133385755171     20090608     20090608   
2359246  7E773C5E07AACB89  887793388336172     20101213     20101213   
2359806  7E81F53424A3B00C  887453385654794     20081205     20081205   
2360000  7E860FB9345DAF65  887073385991119     20081003     20081003   
2361569  7E9875157A915F8A  887083386304533     20100213     20100213   
2363074  7EAEC15E06D9B685  887993385782038     20101012     20101012   
2363373  7EB4BD1BB601E201  887523385967880     20090102     20090102   
2363468  7EB6D72572D996DB  887463386650590     20091206     20091206   
2363855  7EBD1CA770D1215F  887453388090084     20090701     20090701   
2364135  7EC03B321A9E319F  887213388754063     20101103     20101103   
2364887  7EC637BC6A7EFF9E  887143384989589     20090115     20090117   
2365312  7ECE370A4F03BB55  887873385114592     20100211     20100211   
2365539  7ED2D662722FE5C3  887213388143889     20100216     20100216   
2365765  7ED5D270889285D9  887723385345772     20090618     20090618   
2365997  7ED9E4B16B37DDAC  887783388937021     20090912     20090912   
2366104  7EDD7AA045A115E5  887523384951443     20080905     20080905   
2367357  7EEE09EC27605873  887303388266352     20080127     20080127   
2367597  7EF3D29C645FF758  887093386485165     20090111     20090111   
2367737  7EF534D9A8000141  887743387774513     20090712     20090712   
2367779  7EF639D1344E5ED6  887963388352341     20080129     20080129   
2368096  7EFD0BC2354BB90D  887113388596111     20090331     20090331   
2368132  7EFD6371947E6B20  887193387952900     20080202     20080202   
2368427  7EFEE535C17CF51A  887903387227929     2009010

> what is the count of various 599 prefixes? 

In [13]:
new = utifive.groupby("ICD9_DGNS_CD_1")["ICD9_DGNS_CD_1"].count()
print (new)

ICD9_DGNS_CD_1
5990     1082
5991      111
5992      106
5993      112
5994      119
5995      116
5996      333
59960     380
59969     316
5997      377
59970     357
59971     342
59972     350
5998      325
59981     104
59982     116
59983     103
59984     121
59989     358
5999      346
Name: ICD9_DGNS_CD_1, dtype: int64


![](https://i.imgur.com/wHBeC7X.png) (according to [this site](https://dexur.com/icd9/599/)) 

# ...⇧
| FINDING: we want to focus on **5990**